### PR TITLE
[Runtimes] Avoid hang on large stderr output in `run_exec`

### PR DESCRIPTION
**Pull request by gtopper**
[ML-3710](https://jira.iguazeng.com/browse/ML-3710)

### CODE REVIEW

**Review by liranbg — CHANGES_REQUESTED** [review]
add a ut that covers writing >=8k to stderr

**Review by liranbg — APPROVED** [review]

File: mlrun/runtimes/local.py [hunk 1]
```diff
@@ -362,17 +363,33 @@ def run_exec(cmd, args, env=None, cwd=None):
...
+    def read_stderr(stderr):
+        while True:
+            nextline = process.stderr.readline()
+            if not nextline:
+                break
+            stderr[0] += nextline
+
+    # ML-3710. We must read stderr in a separate thread to drain the stderr pipe so that the spawned process won't
+    # hang if it tries to write more to stderr than the buffer size (default of approx 8kb).
+    stderr = [""]
```
Comment (assaf758) [inline mlrun/runtimes/local.py:379]: why stderr has to be a list and not an str?

Comment (gtopper) [reply mlrun/runtimes/local.py]: To get a reference type for inter-thread communication, but now using `StringIO` so list is no longer necessary.

File: tests/runtimes/test_local.py [hunk 2]
```diff
@@ -23,6 +24,9 @@ def test_run_exec_basic():
 
 # ML-3710
 def test_run_exec_verbose_stderr():
-    out, err = run_exec(["python"], ["assets/verbose_stderr.py"])
+    script_path = str(pathlib.Path(__file__).parent / "assets" / "verbose_stderr.py")
+    out, err = run_exec(["python"], [script_path])
     assert out == "some output\n"
+    print(f"out={out}")
+    print(f"err={err}")
```
Comment (assaf758) [inline tests/runtimes/test_local.py:31]: i guess would spam the CI..

Comment (gtopper) [reply tests/runtimes/test_local.py:31]: Oops, debugging...

**Review by assaf758 — APPROVED** [review]

**Comment by liranbg** [comment]
@gtopper  - can be 1.3.1 candidate?

**Pull request merged**